sdk/core/src/migration/credentials/index.ts

```diff
@@ -18,91 +10,33 @@ interface vcMigrationList {
-  private auth: string
-  private keysService: KeysService
-  private platformCryptographyTools: IPlatformCryptographyTools
-  private readonly apiKey: string
-  private readonly api: ApiService
-  private tokenRequestTime: number
-
-  constructor(
-    didAuthAdapter: DidAuthAdapter,
-    apiKey: string,
-    keysService: KeysService,
-    platformCryptographyTools: IPlatformCryptographyTools,
-    private readonly bloomDid: string,
-  ) {
-    this.baseUrl = VAULT_MIGRATION_SERVICE_URL
-    this.apiKey = apiKey
-    this.didAuthService = new DidAuthService(didAuthAdapter, this.apiKey, this.baseUrl)
-    this.keysService = keysService
-    this.platformCryptographyTools = platformCryptographyTools
-    this.api = new ApiService(this.baseUrl, {
-      'Api-Key': apiKey,
-      'X-SDK-Version': version,
+  private readonly migrationApiService
+  private readonly encryptionService
+  private readonly bloomDid
+
+  constructor({ accessApiKey, bloomDid, didAuthAdapter, encryptionService, migrationUrl }: ConstructorOptions) {
+    this.migrationApiService = new VaultMigrationApiService({
+      accessApiKey,
+      didAuthAdapter,
+      migrationUrl,
+      sdkVersion: extractSDKVersion(),
     })
-  }
-
-  /**
-   * Fetches authorization token to `vault-migration-service`
-   */
-  private async getAuth(): Promise<string> {
-    if (this.auth && !this.didAuthService.isTokenExpired(this.auth, this.tokenRequestTime)) {
-      return this.auth
-    }
-
-    this.tokenRequestTime = Date.now()
-    const requestToken = await this.didAuthService.pullDidAuthRequestToken()
-    this.auth = await this.didAuthService.createDidAuthResponseToken(requestToken)
-
-    return this.auth
-  }
-
-  /**
-   * Copy of the private method of `AffinidiVaultStorageService`
-   * @param credentials
-   * @private
-   */
-  async encryptCredentials(credentials: any[]): Promise<vcMigrationList[]> {
-    const publicKeyBuffer = this.keysService.getOwnPublicKey()
-    const privateKeyBuffer = this.keysService.getOwnPrivateKey()
-    const encryptedCredentials: vcMigrationList[] = []
-
-    for (const credential of credentials) {
-      let credentialId = credential?.id
-      if (!isW3cCredential(credential)) {
-        credentialId = credential?.data?.id
-      }
-
-      const credentialIdHash = await this.platformCryptographyTools.computePersonalHash(privateKeyBuffer, credentialId)
-
-      const typeHashes = []
-      if (isW3cCredential(credential)) {
-        for (const credentialType of credential.type) {
-          const typeHash = await this.platformCryptographyTools.computePersonalHash(privateKeyBuffer, credentialType)
-          typeHashes.push(typeHash)
-        }
-      }
-
-      const cyphertext = await this.platformCryptographyTools.encryptByPublicKey(publicKeyBuffer, credential)
-
-      encryptedCredentials.push({
-        id: credentialIdHash,
-        types: typeHashes,
-        payload: cyphertext,
-        bloomVaultIndex: credential.bloomId,
-      })
```

Comment:
previously we had contract with VMS for such properties and now that is breaking as you are using same method as for AffinidiVaultStorageService.

Reply:
Because comment for this method (on line 70) says it's "Copy of the private method of `AffinidiVaultStorageService`"?

But apparently it's not just a copy after all; I'll fix it.

Reply:
@inga-affinidi I wrote that comment "Copy of the private method of `AffinidiVaultStorageService`" because I wanted to highlight that it is the same **encryption** implementation as in `AffinidiVaultStorageService` and we can decrypt messages inside `affinidi-vault` using its own decryption methods.

The difference is in the additional field `bloomVaultIndex` that we need for migration process.

Reply:
On the second thought, why do integration tests pass then?

Reply:
@inga-affinidi We have integration tests covering only migration endpoints themselves. Another functionality was covered with unit tests.

Reply:
@YuriiSkrypnyk , why did you approve this PR then? :)

@YuriiSkrypnyk , @oleksiipiliugin , fixed this issue